### PR TITLE
Add schedule detail to job list expanded view

### PR DESCRIPTION
**Pull request by marshmalien**
##### SUMMARY
Issue: https://github.com/ansible/awx/issues/8321

Screen Shot 2022-02-22 at 3 29 10 PM

##### ISSUE TYPE

 - Enhancement Pull Request

##### COMPONENT NAME

 - UI

### CODE REVIEW

File: awx/ui/src/components/DetailList/LaunchedByDetail.js [hunk 1]
```diff
@@ -1,45 +1,16 @@
 import React from 'react';
 import { Link } from 'react-router-dom';
 import { t } from '@lingui/macro';
+import getScheduleUrl from 'util/getScheduleUrl';
```
Comment (AlexSCorey) [inline awx/ui/src/components/DetailList/LaunchedByDetail.js:4]: I like the idea of moving out this of this file.

File: awx/ui/src/components/JobList/JobListItem.js [hunk 2]
```diff
@@ -167,6 +168,17 @@ function JobListItem({
                   />
                 )}
               <LaunchedByDetail job={job} />
+              {job.launch_type === 'scheduled' && (
+                <Detail
+                  dataCy="job-schedule"
+                  label={t`Schedule`}
+                  value={
+                    <Link to={getScheduleUrl(job)}>
+                      {job.summary_fields.schedule.name}
```
Comment (AlexSCorey) [inline awx/ui/src/components/JobList/JobListItem.js]: You could add `schedule` to the destructed job object [here](https://github.com/ansible/awx/pull/11792/files#diff-04d85be33263d77d4c6a8ba191660b3524a19d0b96d788ded224e23484693551R45-R54)

File: awx/ui/src/components/JobList/JobListItem.test.js [hunk 3]
```diff
@@ -51,7 +60,11 @@ describe('<JobListItem />', () => {
 
   test('initially renders successfully', () => {
     expect(wrapper.find('JobListItem').length).toBe(1);
+  });
```
Comment (AlexSCorey) [inline awx/ui/src/components/JobList/JobListItem.test.js:63]: Do we assert anywhere that the Scheduled value does not show up when it shouldn't, if not we probably should.

**Review by AlexSCorey — COMMENTED** [review]
This looks really good to me.  My only real concern is that we write tests to assert the schedule does not show up when it shouldn't.  Everything else is just my thoughts.

File: awx/ui/src/util/getScheduleUrl.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+export default function getScheduleUrl(job) {
+  const templateId = job.summary_fields.unified_job_template.id;
+  const scheduleId = job.summary_fields.schedule.id;
```
Comment (nixocio) [inline awx/ui/src/util/getScheduleUrl.js:3]: hey @marshmalien, good job. I added 2 schedules to job template, and the job details showed the schedule what triggered the job. However if I delete those schedules, I got the following error:

image

Once the error above is fixed. We get an empty schedule.  For other items deleted, we add a warning - `Resource deleted`. Perhaps we could add some similar.

image

Comment (nixocio) [inline awx/ui/src/util/getScheduleUrl.js:3]:
```suggestion
  const scheduleId = job.summary_fields?.schedule.id;
```

File: awx/ui/src/components/JobList/JobListItem.js (hunk 2 above)
Comment (nixocio) [inline awx/ui/src/components/JobList/JobListItem.js]:
```suggestion
                      {job.summary_fields.schedule?.name}
```

File: awx/ui/src/components/JobList/JobListItem.test.js (hunk 3 above)
Comment (marshmalien) [reply awx/ui/src/components/JobList/JobListItem.test.js:63]: 👍 Added a new test that checks for this case "should not show schedule detail in expanded view".

File: awx/ui/src/util/getScheduleUrl.js (hunk 4 above)
Comment (marshmalien) [reply awx/ui/src/util/getScheduleUrl.js:3]: Updated PR to handle Deleted schedules.
Screen Shot 2022-02-28 at 3 51 07 PM

**Review by nixocio — APPROVED** [review]

**Review by AlexSCorey — APPROVED** [review]
Well done.

**Review by akus062381 — APPROVED** [review]

**Pull request merged**